### PR TITLE
Update flannel to v0.25.4 and fixed issue with IPv6 mask

### PR DESCRIPTION
**Pull request by rbrtbnfgl**
#### Proposed Changes ####

Bump Flannel version to v0.25.4 and fixes an issue when IPv6masq flag is configured
#### Linked Issues ####

#10419

#### User-Facing Change ####

```release-note

```

### CODE REVIEW

**Review by brandond — APPROVED** [review]

File: pkg/agent/flannel/flannel.go [hunk 1]
```diff
@@ -228,34 +230,73 @@ func WriteSubnetFile(path string, nw ip.IP4Net, nwv6 ip.IP6Net, ipMasq bool, bn
...
 		} else if prevCIDRString, ok := prevSubnetVals[CIDRKey]; ok {
-			err = prevCIDR.UnmarshalJSON([]byte(prevCIDRString))
-			if err != nil {
-				logrus.Errorf("Couldn't parse previous %s from subnet file at %s: %v", CIDRKey, path, err)
+			cidrs := strings.Split(prevCIDRString, ",")
+			prevCIDRs = make([]ip.IP4Net, 0)
+			for i := range cidrs {
+				_, cidr, err := net.ParseCIDR(cidrs[i])
+				if err != nil {
+					logrus.Errorf("Couldn't parse previous %s from subnet file at %s: %s", CIDRKey, path, err)
```
Comment (manuelbuil) [inline pkg/agent/flannel/flannel.go:257]: same

File: pkg/agent/flannel/flannel.go [hunk 2]
```diff
@@ -228,34 +230,73 @@ func WriteSubnetFile(path string, nw ip.IP4Net, nwv6 ip.IP6Net, ipMasq bool, bn
...
+	}
+}
+
+func ReadIP6CIDRsFromSubnetFile(path string, CIDRKey string) []ip.IP6Net {
+	prevCIDRs := make([]ip.IP6Net, 0)
 	if _, err := os.Stat(path); !os.IsNotExist(err) {
 		prevSubnetVals, err := godotenv.Read(path)
 		if err != nil {
-			logrus.Errorf("Couldn't fetch previous %s from subnet file at %s: %v", CIDRKey, path, err)
+			logrus.Errorf("Couldn't fetch previous %s from subnet file at %s: %s", CIDRKey, path, err)
```
Comment (manuelbuil) [inline pkg/agent/flannel/flannel.go:287]: same

**Review by manuelbuil — CHANGES_REQUESTED** [review]

File: pkg/agent/flannel/flannel.go [hunk 3]
```diff
@@ -228,34 +230,73 @@ func WriteSubnetFile(path string, nw ip.IP4Net, nwv6 ip.IP6Net, ipMasq bool, bn
...
 		} else if prevCIDRString, ok := prevSubnetVals[CIDRKey]; ok {
-			err = prevCIDR.UnmarshalJSON([]byte(prevCIDRString))
-			if err != nil {
-				logrus.Errorf("Couldn't parse previous %s from subnet file at %s: %v", CIDRKey, path, err)
+			cidrs := strings.Split(prevCIDRString, ",")
+			prevCIDRs = make([]ip.IP6Net, 0)
+			for i := range cidrs {
+				_, cidr, err := net.ParseCIDR(cidrs[i])
+				if err != nil {
+					logrus.Errorf("Couldn't parse previous %s from subnet file at %s: %s", CIDRKey, path, err)
```
Comment (manuelbuil) [inline pkg/agent/flannel/flannel.go:294]: same here, sorry I missed it

**Review by manuelbuil — CHANGES_REQUESTED** [review]
The problem is that it reads the subnet file, e.g.
```
FLANNEL_NETWORK=10.42.0.0/16
FLANNEL_SUBNET=10.42.0.1/24
FLANNEL_IPV6_NETWORK=2001:cafe:42::/56
FLANNEL_IPV6_SUBNET=2001:cafe:42::1/64
FLANNEL_MTU=1450
FLANNEL_IPMASQ=true
```
But there are two IPv6 networks?

**Review by dereknola — APPROVED** [review]

**Pull request merged**